### PR TITLE
feat: declare event logger manager and public APIs (SDKCF-6845)

### PR DESCRIPTION
**Pull request by Esakkiraja-Pothikannan**
- Added public facing event logger manager class
- Declared public APIs 
- Changed some device info variables as non-optional

Link: 
SDKCF-6845

### CODE REVIEW

**Review by SoumenRautray — APPROVED** [review]

File: Sources/REventLogger/REvent.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    let osVer: String
+    let deviceModel: String
+    let deviceBrand: String
+    let deviceName: String
+    let sourceName: String
+    let sourceVer: String
+    let errorCode: String
+    let errorMsg: String
+    var eventVer: String = "1"
+    var platform: String = "iOS"
```
Comment (anish-munirathinam) [inline Sources/REventLogger/REvent.swift:17]: why `var`? should it be `mutable`?

File: Sources/REventLogger/REventLogger.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+struct EventLoggerConfiguration {
+    let apiKey: String
+    let apiUrl: String
+}
+
+/// Event Logger that sends the custom events to the Event Logger Service
+public final class REventLogger {
+    /// Singleton shared instance of REventLogger
+    public static let shared = REventLogger()
+    static var configration: EventLoggerConfiguration?
```
Comment (anish-munirathinam) [inline Sources/REventLogger/REventLogger.swift:17]: typo `configuration`.

File: Sources/REventLogger/REventLogger.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import Foundation
+
+/// Descripe the type of the event
```
Comment (anish-munirathinam) [inline Sources/REventLogger/REventLogger.swift:3]: typo `describe`.

File: Sources/REventLogger/REvent.swift [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    var occurrenceCount: Int = 0
+    var firstOccurrenceOn: Double // unix time
```
Comment (anish-munirathinam) [inline Sources/REventLogger/REvent.swift:19]: nit: could be renamed to `firstOccurrenceTimestamp`.

Comment (maureenorea-clores) [reply Sources/REventLogger/REvent.swift:19]: @anish-munirathinam @Esakkiraja-Pothikannan I prefer to have the unit specified to know just by looking at the request. e.g. `firstOccurrenceMillis` or `firstOccurrenceMs`.`Timestamp` can still be a bit ambiguous. What do you think?

Comment (maureenorea-clores) [reply Sources/REventLogger/REvent.swift:19]: As long as both platforms are consistent it is fine. Let's stick with `firstOccurrenceOn` then.

File: Tests/Tests/REventLogger/REventSpec.swift [hunk 5]
```diff
@@ -0,0 +1,8 @@
+//
+//  File.swift
+//  
+//
+//  Created by P, Esakkiraja on 27/11/23.
+//
+
+import Foundation
```
Comment (anish-munirathinam) [inline Tests/Tests/REventLogger/REventSpec.swift:8]: These files are just created but no tests are added yet.

File: Sources/REventLogger/REventLogger.swift [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                                         sourceVersion: String,
+                                         errorCode: String,
+                                         errorMessage: String,
+                                         info: [String: String]? = nil) {
+        // Send the event to server
+        // Save it in the local DB, as warning
+    }
+
+    /// Logs the warning event
+    /// This event will be considered as low priority and will be sent with bacth update.
```
Comment (anish-munirathinam) [inline Sources/REventLogger/REventLogger.swift:50]: nit: typo as `batch` updates.

File: Tests/Tests/REventLogger/REventSpec.swift (hunk 5 above)
Comment (anish-munirathinam) [reply Tests/Tests/REventLogger/REventSpec.swift:8]: Better to remove these files, if it is empty for now.

**Pull request merged**